From 110a349e200bc9e323108b3614a8fdbdbb5551f8 Mon Sep 17 00:00:00 2001
From: shengfei Xu <xsf@rock-chips.com>
Date: Fri, 20 Mar 2020 10:06:09 +0800
Subject: [PATCH] ARM: dts: rv1126: Set pwm pin default pull setting when used
 for pwm regulator

As a second global reset, the GRF is not reset, the iomux and
pull of PWM pin is still keeping, but PWM controller is reset,
PWM pin goes into input mode. However, the pull is still none
changed in kernel, which can cause voltage problems, so should
always keep the PWM pin default mode.

Signed-off-by: shengfei Xu <xsf@rock-chips.com>
Change-Id: Ie07eea6a8bb7eec3766244fd3476aff16b9f5b83
---
 arch/arm/boot/dts/rv1126-pinctrl.dtsi | 144 ++++++++++++++++++++++++++
 1 file changed, 144 insertions(+)

diff --git a/arch/arm/boot/dts/rv1126-pinctrl.dtsi b/arch/arm/boot/dts/rv1126-pinctrl.dtsi
index 6717aade32f4..9ad207d8a848 100644
--- a/arch/arm/boot/dts/rv1126-pinctrl.dtsi
+++ b/arch/arm/boot/dts/rv1126-pinctrl.dtsi
@@ -831,11 +831,23 @@
 				<0 RK_PB6 3 &pcfg_pull_none>;
 		};
 		/omit-if-no-ref/
+		pwm0m0_pins_pull_down: pwm0m0-pins-pull-down {
+			rockchip,pins =
+				/* pwm0_m0 */
+				<0 RK_PB6 3 &pcfg_pull_down>;
+		};
+		/omit-if-no-ref/
 		pwm0m1_pins: pwm0m1-pins {
 			rockchip,pins =
 				/* pwm0_m1 */
 				<2 RK_PB3 5 &pcfg_pull_none>;
 		};
+		/omit-if-no-ref/
+		pwm0m1_pins_pull_down: pwm0m1-pins-pull-down {
+			rockchip,pins =
+				/* pwm0_m1 */
+				<2 RK_PB3 5 &pcfg_pull_down>;
+		};
 	};
 	pwm1 {
 		/omit-if-no-ref/
@@ -845,11 +857,23 @@
 				<0 RK_PB7 3 &pcfg_pull_none>;
 		};
 		/omit-if-no-ref/
+		pwm1m0_pins_pull_down: pwm1m0-pins-pull-down {
+			rockchip,pins =
+				/* pwm1_m0 */
+				<0 RK_PB7 3 &pcfg_pull_down>;
+		};
+		/omit-if-no-ref/
 		pwm1m1_pins: pwm1m1-pins {
 			rockchip,pins =
 				/* pwm1_m1 */
 				<2 RK_PB2 5 &pcfg_pull_none>;
 		};
+		/omit-if-no-ref/
+		pwm1m1_pins_pull_down: pwm1m1-pins-pull-down {
+			rockchip,pins =
+				/* pwm1_m1 */
+				<2 RK_PB2 5 &pcfg_pull_down>;
+		};
 	};
 	pwm10 {
 		/omit-if-no-ref/
@@ -859,11 +883,23 @@
 				<3 RK_PA6 6 &pcfg_pull_none>;
 		};
 		/omit-if-no-ref/
+		pwm10m0_pins_pull_down: pwm10m0-pins-pull-down {
+			rockchip,pins =
+				/* pwm10_m0 */
+				<3 RK_PA6 6 &pcfg_pull_down>;
+		};
+		/omit-if-no-ref/
 		pwm10m1_pins: pwm10m1-pins {
 			rockchip,pins =
 				/* pwm10_m1 */
 				<2 RK_PD5 5 &pcfg_pull_none>;
 		};
+		/omit-if-no-ref/
+		pwm10m1_pins_pull_down: pwm10m1-pins-pull-down {
+			rockchip,pins =
+				/* pwm10_m1 */
+				<2 RK_PD5 5 &pcfg_pull_down>;
+		};
 	};
 	pwm11 {
 		/omit-if-no-ref/
@@ -873,11 +909,23 @@
 				<3 RK_PA7 6 &pcfg_pull_none>;
 		};
 		/omit-if-no-ref/
+		pwm11m0_pins_pull_down: pwm11m0-pins-pull-down {
+			rockchip,pins =
+				/* pwm11_ir_m0 */
+				<3 RK_PA7 6 &pcfg_pull_down>;
+		};
+		/omit-if-no-ref/
 		pwm11m1_pins: pwm11m1-pins {
 			rockchip,pins =
 				/* pwm11_ir_m1 */
 				<2 RK_PD4 5 &pcfg_pull_none>;
 		};
+		/omit-if-no-ref/
+		pwm11m1_pins_pull_down: pwm11m1-pins-pull-down {
+			rockchip,pins =
+				/* pwm11_ir_m1 */
+				<2 RK_PD4 5 &pcfg_pull_down>;
+		};
 	};
 	pwm2 {
 		/omit-if-no-ref/
@@ -887,11 +935,23 @@
 				<0 RK_PC0 3 &pcfg_pull_none>;
 		};
 		/omit-if-no-ref/
+		pwm2m0_pins_pull_down: pwm2m0-pins-pull-down {
+			rockchip,pins =
+				/* pwm2_m0 */
+				<0 RK_PC0 3 &pcfg_pull_down>;
+		};
+		/omit-if-no-ref/
 		pwm2m1_pins: pwm2m1-pins {
 			rockchip,pins =
 				/* pwm2_m1 */
 				<2 RK_PB1 5 &pcfg_pull_none>;
 		};
+		/omit-if-no-ref/
+		pwm2m1_pins_pull_down: pwm2m1-pins-pull-down {
+			rockchip,pins =
+				/* pwm2_m1 */
+				<2 RK_PB1 5 &pcfg_pull_down>;
+		};
 	};
 	pwm3 {
 		/omit-if-no-ref/
@@ -901,11 +961,23 @@
 				<0 RK_PC1 3 &pcfg_pull_none>;
 		};
 		/omit-if-no-ref/
+		pwm3m0_pins_pull_down: pwm3m0-pins-pull-down {
+			rockchip,pins =
+				/* pwm3_ir_m0 */
+				<0 RK_PC1 3 &pcfg_pull_down>;
+		};
+		/omit-if-no-ref/
 		pwm3m1_pins: pwm3m1-pins {
 			rockchip,pins =
 				/* pwm3_ir_m1 */
 				<2 RK_PB0 5 &pcfg_pull_none>;
 		};
+		/omit-if-no-ref/
+		pwm3m1_pins_pull_down: pwm3m1-pins-pull-down {
+			rockchip,pins =
+				/* pwm3_ir_m1 */
+				<2 RK_PB0 5 &pcfg_pull_down>;
+		};
 	};
 	pwm4 {
 		/omit-if-no-ref/
@@ -915,11 +987,23 @@
 				<0 RK_PC2 3 &pcfg_pull_none>;
 		};
 		/omit-if-no-ref/
+		pwm4m0_pins_pull_down: pwm4m0-pins-pull-down {
+			rockchip,pins =
+				/* pwm4_m0 */
+				<0 RK_PC2 3 &pcfg_pull_down>;
+		};
+		/omit-if-no-ref/
 		pwm4m1_pins: pwm4m1-pins {
 			rockchip,pins =
 				/* pwm4_m1 */
 				<2 RK_PA7 5 &pcfg_pull_none>;
 		};
+		/omit-if-no-ref/
+		pwm4m1_pins_pull_down: pwm4m1-pins-pull-down {
+			rockchip,pins =
+				/* pwm4_m1 */
+				<2 RK_PA7 5 &pcfg_pull_down>;
+		};
 	};
 	pwm5 {
 		/omit-if-no-ref/
@@ -929,11 +1013,23 @@
 				<0 RK_PC3 3 &pcfg_pull_none>;
 		};
 		/omit-if-no-ref/
+		pwm5m0_pins_pull_down: pwm5m0-pins-pull-down {
+			rockchip,pins =
+				/* pwm5_m0 */
+				<0 RK_PC3 3 &pcfg_pull_down>;
+		};
+		/omit-if-no-ref/
 		pwm5m1_pins: pwm5m1-pins {
 			rockchip,pins =
 				/* pwm5_m1 */
 				<2 RK_PA6 5 &pcfg_pull_none>;
 		};
+		/omit-if-no-ref/
+		pwm5m1_pins_pull_down: pwm5m1-pins-pull-down {
+			rockchip,pins =
+				/* pwm5_m1 */
+				<2 RK_PA6 5 &pcfg_pull_down>;
+		};
 	};
 	pwm6 {
 		/omit-if-no-ref/
@@ -943,11 +1039,23 @@
 				<0 RK_PB2 3 &pcfg_pull_none>;
 		};
 		/omit-if-no-ref/
+		pwm6m0_pins_pull_down: pwm6m0-pins-pull-down {
+			rockchip,pins =
+				/* pwm6_m0 */
+				<0 RK_PB2 3 &pcfg_pull_down>;
+		};
+		/omit-if-no-ref/
 		pwm6m1_pins: pwm6m1-pins {
 			rockchip,pins =
 				/* pwm6_m1 */
 				<3 RK_PA1 5 &pcfg_pull_none>;
 		};
+		/omit-if-no-ref/
+		pwm6m1_pins_pull_up: pwm6m1-pins-pull-up {
+			rockchip,pins =
+				/* pwm6_m1 */
+				<3 RK_PA1 5 &pcfg_pull_up>;
+		};
 	};
 	pwm7 {
 		/omit-if-no-ref/
@@ -957,11 +1065,23 @@
 				<0 RK_PB1 3 &pcfg_pull_none>;
 		};
 		/omit-if-no-ref/
+		pwm7m0_pins_pull_down: pwm7m0-pins-pull-down {
+			rockchip,pins =
+				/* pwm7_ir_m0 */
+				<0 RK_PB1 3 &pcfg_pull_down>;
+		};
+		/omit-if-no-ref/
 		pwm7m1_pins: pwm7m1-pins {
 			rockchip,pins =
 				/* pwm7_ir_m1 */
 				<3 RK_PA0 5 &pcfg_pull_none>;
 		};
+		/omit-if-no-ref/
+		pwm7m1_pins_pull_up: pwm7m1-pins-pull-up {
+			rockchip,pins =
+				/* pwm7_ir_m1 */
+				<3 RK_PA0 5 &pcfg_pull_up>;
+		};
 	};
 	pwm8 {
 		/omit-if-no-ref/
@@ -971,11 +1091,23 @@
 				<3 RK_PA4 6 &pcfg_pull_none>;
 		};
 		/omit-if-no-ref/
+		pwm8m0_pins_pull_down: pwm8m0-pins-pull-down {
+			rockchip,pins =
+				/* pwm8_m0 */
+				<3 RK_PA4 6 &pcfg_pull_down>;
+		};
+		/omit-if-no-ref/
 		pwm8m1_pins: pwm8m1-pins {
 			rockchip,pins =
 				/* pwm8_m1 */
 				<2 RK_PD7 5 &pcfg_pull_none>;
 		};
+		/omit-if-no-ref/
+		pwm8m1_pins_pull_down: pwm8m1-pins-pull-down {
+			rockchip,pins =
+				/* pwm8_m1 */
+				<2 RK_PD7 5 &pcfg_pull_down>;
+		};
 	};
 	pwm9 {
 		/omit-if-no-ref/
@@ -985,11 +1117,23 @@
 				<3 RK_PA5 6 &pcfg_pull_none>;
 		};
 		/omit-if-no-ref/
+		pwm9m0_pins_pull_down: pwm9m0-pins-pull-down {
+			rockchip,pins =
+				/* pwm9_m0 */
+				<3 RK_PA5 6 &pcfg_pull_down>;
+		};
+		/omit-if-no-ref/
 		pwm9m1_pins: pwm9m1-pins {
 			rockchip,pins =
 				/* pwm9_m1 */
 				<2 RK_PD6 5 &pcfg_pull_none>;
 		};
+		/omit-if-no-ref/
+		pwm9m1_pins_pull_down: pwm9m1-pins-pull-down {
+			rockchip,pins =
+				/* pwm9_m1 */
+				<2 RK_PD6 5 &pcfg_pull_down>;
+		};
 	};
 	rgmii {
 		/omit-if-no-ref/
-- 
2.35.3

